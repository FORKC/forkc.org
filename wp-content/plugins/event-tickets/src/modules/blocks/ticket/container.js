/**
 * External dependencies
 */
import { connect } from 'react-redux';
import { compose } from 'redux';

/**
 * Internal dependencies
 */
import Template from './template';
import { plugins } from '@moderntribe/common/data';
import { withSaveData, withStore } from '@moderntribe/common/hoc';
import { actions, selectors } from '@moderntribe/tickets/data/blocks/ticket';
import {
	isModalShowing,
	getModalTicketId,
} from '@moderntribe/tickets/data/shared/move/selectors';

const mapStateToProps = ( state, ownProps ) => {
	const props = { blockId: ownProps.clientId };

	return {
		blockId: ownProps.clientId,
		hasTicketsPlus: plugins.selectors.hasPlugin( state )( plugins.constants.TICKETS_PLUS ),
		hasBeenCreated: selectors.getTicketHasBeenCreated( state, props ),
		isDisabled: selectors.isTicketDisabled( state, props ),
		isLoading: selectors.getTicketIsLoading( state, props ),
		ticketId: selectors.getTicketId( state, props ),
		isModalShowing: isModalShowing( state ),
		modalTicketId: getModalTicketId( state ),
	};
};

const mapDispatchToProps = ( dispatch, ownProps ) => {
	const { clientId } = ownProps;

	return {
		onBlockUpdate: ( isSelected ) => (
			dispatch( actions.setTicketIsSelected( clientId, isSelected ) )
		),
		setInitialState: ( props ) => {
			dispatch( actions.registerTicketBlock( clientId ) );
			dispatch( actions.setTicketInitialState( props ) );
		},
	};
};

const mergeProps = ( stateProps, dispatchProps, ownProps ) => ( {
	...stateProps,
	...dispatchProps,
	...ownProps,
	isModalShowing: stateProps.isModalShowing && stateProps.modalTicketId === stateProps.ticketId,
} );

export default compose(
	withStore( { isolated: true } ),
	connect(
		mapStateToProps,
		mapDispatchToProps,
		mergeProps,
	),
	withSaveData(),
)( Template );

