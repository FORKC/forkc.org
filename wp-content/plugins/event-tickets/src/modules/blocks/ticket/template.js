/**
 * External dependencies
 */
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

/**
 * WordPress dependencies
 */
import { Spinner } from '@wordpress/components';

/**
 * Internal dependencies
 */
import './style.pcss';
import TicketContainer from './container/container';
import TicketDashboard from './dashboard/container';
import MoveModal from '@moderntribe/tickets/elements/move-modal';

class Ticket extends PureComponent {
	static propTypes = {
		blockId: PropTypes.string.isRequired,
		hasTicketsPlus: PropTypes.bool,
		isDisabled: PropTypes.bool,
		isLoading: PropTypes.bool,
		isModalShowing: PropTypes.bool,
		isSelected: PropTypes.bool,
		onBlockUpdate: PropTypes.func,
		removeTicketBlock: PropTypes.func,
	};

	componentDidMount() {
		this.props.onBlockUpdate( this.props.isSelected );
	}

	componentDidUpdate( prevProps ) {
		if ( prevProps.isSelected !== this.props.isSelected ) {
			this.props.onBlockUpdate( this.props.isSelected );
		}
	}

	render() {
		const {
			blockId,
			hasTicketsPlus,
			isDisabled,
			isLoading,
			isSelected,
			isModalShowing,
		} = this.props;

		return [
			(
				<article className={ classNames(
					'tribe-editor__ticket',
					{ 'tribe-editor__ticket--disabled': isDisabled },
					{ 'tribe-editor__ticket--selected': isSelected },
					{ 'tribe-editor__ticket--has-tickets-plus': hasTicketsPlus },
				) }
				>
					<TicketContainer blockId={ blockId } isSelected={ isSelected } />
					<TicketDashboard blockId={ blockId } isSelected={ isSelected } />
					{ isLoading && <Spinner /> }
				</article>
			),
			isModalShowing && <MoveModal />,
		];
	}
}

export default Ticket;
