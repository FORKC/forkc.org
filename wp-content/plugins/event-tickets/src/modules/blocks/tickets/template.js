/**
 * External dependencies
 */
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

/**
 * Internal dependencies
 */
import TicketsDashboard from './dashboard/container';
import TicketsContainer from './container/container';
import TicketControls from './controls/container';
import './style.pcss';

class Tickets extends PureComponent {
	static propTypes = {
		isSelected: PropTypes.bool,
		canCreateTickets: PropTypes.bool,
		isSettingsOpen: PropTypes.bool,
		clientId: PropTypes.string,
		header: PropTypes.string,
	};

	render() {
		const {
			isSelected,
			canCreateTickets,
			isSettingsOpen,
			clientId,
		} = this.props;

		return (
			<div
				className={ classNames(
					'tribe-editor__tickets',
					{ 'tribe-editor__tickets--selected': isSelected },
					{ 'tribe-editor__tickets--settings-open': isSettingsOpen },
				) }
			>
				<TicketsContainer isSelected={ isSelected } />
				{ canCreateTickets && <TicketsDashboard isSelected={ isSelected } clientId={ clientId } /> }
				<TicketControls />
			</div>
		);
	}
}

export default Tickets;
