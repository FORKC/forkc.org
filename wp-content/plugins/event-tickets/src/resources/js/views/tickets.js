// @TODO: Take this line off once we actually have the tribe object
if ( 'undefined' === typeof tribe ) {
	tribe = {};
}

// Define the tickets object if not defined
if ( 'undefined' === typeof tribe.tickets ) {
	tribe.tickets = {};
}

tribe.tickets.block = {
	num_attendees: 0,
	event        : {}
};

( function( $, obj ) {
	'use strict';

	obj.selector = {
		container                 : '.tribe-block__tickets',
		submit                    : '.tribe-block__tickets__buy',
		item                      : '.tribe-block__tickets__item',
		itemOptOut                : '.tribe-block__tickets__item__optout',
		itemQuantity              : '.tribe-block__tickets__item__quantity',
		itemQuantityInput         : '.tribe-ticket-quantity',
		itemExtraAvailable        : '.tribe-block__tickets__item__extra__available',
		itemExtraAvailableQuantity: '.tribe-block__tickets__item__extra__available_quantity'
	};

	var $tribe_ticket = $( obj.selector.container );

	// Bail if there are no tickets on the current event/page/post
	if ( 0 === $tribe_ticket.length ) {
		return;
	}

	/**
	 * Handle the number input + and - actions
	 *
	 * @since 4.9
	 *
	 * @return void
	 */
	$( obj.selector.item ).on( 'click',
		'.tribe-block__tickets__item__quantity__remove, .tribe-block__tickets__item__quantity__add',
		function( e ) {
			e.preventDefault();
			var input  = $( this ).parent().find( 'input[type="number"]' );
			var add    = $( this ).hasClass( 'tribe-block__tickets__item__quantity__add' );

			// stepUp or stepDown the input according to the button that was clicked
			add ? input[0].stepUp() : input[0].stepDown();

			// Trigger the on Change for the input as it's not handled via stepUp() || stepDown()
			input.trigger( 'change' );

	} );

	/**
	 * Handle the TPP form
	 *
	 * @since 4.9
	 *
	 * @return void
	 */
	$( obj.selector.item ).on( 'change',
		'.tribe-ticket-quantity',
		function( e ) {
			var $this      = $( this );
			var $ticket    = $this.closest( obj.selector.item );
			var $ticket_id = $ticket.data( 'ticket-id' );

			var $form = $this.closest( obj.selector.container );

			var new_quantity = parseInt( $this.val(), 10 );
			new_quantity     = isNaN( new_quantity ) ? 0 : new_quantity;

			// Maybe display the Opt Out
			var $has_optout = $ticket.has( obj.selector.itemOptOut ).length;
			if ( $has_optout ) {
				( new_quantity > 0 ) ? $( obj.selector.itemOptOut ).show() : $( obj.selector.itemOptOut ).hide();
			}

			// Only disable / enable if is a Tribe Commerce Paypal form.
			if ( 'Tribe__Tickets__Commerce__PayPal__Main' !== $form.data( 'provider' ) ) {
				return;
			}

			if ( new_quantity > 0 ) {
				$form
					.find( '[data-ticket-id]:not([data-ticket-id="' + $ticket_id + '"])' )
					.closest( 'div.tribe-block__tickets__item' )
					.find( 'input, button' )
					.attr( 'disabled', 'disabled' )
					.closest( 'div' )
					.addClass( 'tribe-tickets-purchase-disabled' );

			} else {
				$form
					.find( 'input, button' )
					.removeAttr( 'disabled' )
					.closest( 'div' )
					.removeClass( 'tribe-tickets-purchase-disabled' );
			}

	} );

	/**
	 * Get the tickets IDs
	 *
	 * @since 4.9
	 *
	 * @return array
	 */
	obj.getTickets = function() {

		var $tickets = $( obj.selector.item ).map( function() {
			return $( this ).data( 'ticket-id' );
		} ).get();

		return $tickets;
	}

	/**
	 * Make dom updates for the AJAX response
	 *
	 * @since 4.9
	 *
	 * @return array
	 */
	obj.updateAvailability = function( tickets ) {

		Object.keys( tickets ).forEach( function( ticket_id ) {

			var available = tickets[ ticket_id ].available;
			var $ticketEl = $( obj.selector.item + "[data-ticket-id='" + ticket_id + "']" );

			if ( 0 === available ) { // ticket is out of stock

				var unavailableHtml = tickets[ ticket_id ].unavailable_html;
				// Set the availability data attribute to false
				$ticketEl.attr( 'available', false );
				// Remove classes for instock and purchasable
				$ticketEl.removeClass( 'instock' );
				$ticketEl.removeClass( 'purchasable' );

				// Update HTML elements with the "Out of Stock" messages
				$ticketEl.find( obj.selector.itemExtraAvailable ).replaceWith( unavailableHtml );
				$ticketEl.find( obj.selector.itemQuantity ).html( unavailableHtml );
			}

			if ( 1 < available ) { // Ticket in stock, we may want to update values
				$ticketEl.find( obj.selector.itemQuantityInput ).attr( { 'max' : available } );
				$ticketEl.find( obj.selector.itemExtraAvailableQuantity ).html( available );
			}

		});
	}

	/**
	 * Check tickets availability
	 *
	 * @since 4.9
	 *
	 * @return void
	 */
	obj.checkAvailability = function() {

		// We're checking availability for all the tickets at once
		var params = {
			action  : 'ticket_availability_check',
			tickets : obj.getTickets(),
		};

		$.post(
			TribeTickets.ajaxurl,
			params,
			function( response ) {
				var success = response.success;

				// Bail if we don't get a successful response
				if ( ! success ) {
					return;
				}

				// Get the tickets response with availability
				var tickets = response.data.tickets;

				// Make DOM updates
				obj.updateAvailability( tickets );

			}
		);

		// Repeat every 15 seconds
		setTimeout( obj.checkAvailability, 15000 );

	}

	/**
	 * Init the tickets script
	 *
	 * @since 4.9
	 *
	 * @return void
	 */
	obj.init = function() {
		obj.checkAvailability();
	}

	obj.init();


})( jQuery, tribe.tickets.block );