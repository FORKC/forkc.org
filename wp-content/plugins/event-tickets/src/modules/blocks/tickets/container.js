/**
 * External dependencies
 */
import { connect } from 'react-redux';
import { compose } from 'redux';

/**
 * Internal dependencies
 */
import Template from './template';
import { withSaveData, withStore } from '@moderntribe/common/hoc';
import { actions, selectors } from '@moderntribe/tickets/data/blocks/ticket';

const mapStateToProps = ( state ) => {
	const headerImageId = selectors.getTicketsHeaderImageId( state );
	return {
		header: headerImageId ? `${ headerImageId }` : '',
		hasProviders: selectors.hasTicketProviders(),
		isSettingsOpen: selectors.getTicketsIsSettingsOpen( state ),
		provider: selectors.getTicketsProvider( state ),
		sharedCapacity: selectors.getTicketsSharedCapacity( state ),
		canCreateTickets: selectors.canCreateTickets(),
	};
};

const mapDispatchToProps = ( dispatch ) => ( {
	setInitialState: ( props ) => {
		dispatch( actions.setTicketsInitialState( props ) );
	},
	onBlockUpdate: ( isSelected ) => {
		dispatch( actions.setTicketsIsSelected( isSelected ) );
	},
	onBlockRemoved: () => {
		dispatch( actions.resetTicketsBlock() );
	},
} );

export default compose(
	withStore(),
	connect( mapStateToProps, mapDispatchToProps ),
	withSaveData(),
)( Template );
