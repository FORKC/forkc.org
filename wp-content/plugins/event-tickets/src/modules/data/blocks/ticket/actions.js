/**
 * Internal dependencies
 */
import { types } from '@moderntribe/tickets/data/blocks/ticket';

//
// ─── TICKETS ACTIONS ────────────────────────────────────────────────────────────
//

export const setTicketsInitialState = ( props ) => ( {
	type: types.SET_TICKETS_INITIAL_STATE,
	payload: props,
} );

export const setTicketsHeaderImage = ( payload ) => ( {
	type: types.SET_TICKETS_HEADER_IMAGE,
	payload,
} );

export const setTicketsIsSettingsOpen = ( isSettingsOpen ) => ( {
	type: types.SET_TICKETS_IS_SETTINGS_OPEN,
	payload: {
		isSettingsOpen,
	},
} );

export const setTicketsIsSettingsLoading = ( isSettingsLoading ) => ( {
	type: types.SET_TICKETS_IS_SETTINGS_LOADING,
	payload: {
		isSettingsLoading,
	},
} );

export const openSettings = () => setTicketsIsSettingsOpen( true );
export const closeSettings = () => setTicketsIsSettingsOpen( false );

export const setTicketsProvider = ( provider ) => ( {
	type: types.SET_TICKETS_PROVIDER,
	payload: {
		provider,
	},
} );

export const setTicketsSharedCapacity = ( sharedCapacity ) => ( {
	type: types.SET_TICKETS_SHARED_CAPACITY,
	payload: {
		sharedCapacity,
	},
} );

export const setTicketsTempSharedCapacity = ( tempSharedCapacity ) => ( {
	type: types.SET_TICKETS_TEMP_SHARED_CAPACITY,
	payload: {
		tempSharedCapacity,
	},
} );

//
// ─── HEADER IMAGE SAGA ACTIONS ──────────────────────────────────────────────────
//

export const fetchTicketsHeaderImage = ( id ) => ( {
	type: types.FETCH_TICKETS_HEADER_IMAGE,
	payload: {
		id,
	},
} );

export const updateTicketsHeaderImage = ( image ) => ( {
	type: types.UPDATE_TICKETS_HEADER_IMAGE,
	payload: {
		image,
	},
} );

export const deleteTicketsHeaderImage = () => ( {
	type: types.DELETE_TICKETS_HEADER_IMAGE,
} );

//
// ─── TICKET DETAILS ACTIONS ─────────────────────────────────────────────────────
//

export const setTicketTitle = ( blockId, title ) => ( {
	type: types.SET_TICKET_TITLE,
	payload: {
		blockId,
		title,
	},
} );

export const setTicketDescription = ( blockId, description ) => ( {
	type: types.SET_TICKET_DESCRIPTION,
	payload: {
		blockId,
		description,
	},
} );

export const setTicketPrice = ( blockId, price ) => ( {
	type: types.SET_TICKET_PRICE,
	payload: {
		blockId,
		price,
	},
} );

export const setTicketSku = ( blockId, sku ) => ( {
	type: types.SET_TICKET_SKU,
	payload: {
		blockId,
		sku,
	},
} );

export const setTicketStartDate = ( blockId, startDate ) => ( {
	type: types.SET_TICKET_START_DATE,
	payload: {
		blockId,
		startDate,
	},
} );

export const setTicketStartDateInput = ( blockId, startDateInput ) => ( {
	type: types.SET_TICKET_START_DATE_INPUT,
	payload: {
		blockId,
		startDateInput,
	},
} );

export const setTicketStartDateMoment = ( blockId, startDateMoment ) => ( {
	type: types.SET_TICKET_START_DATE_MOMENT,
	payload: {
		blockId,
		startDateMoment,
	},
} );

export const setTicketEndDate = ( blockId, endDate ) => ( {
	type: types.SET_TICKET_END_DATE,
	payload: {
		blockId,
		endDate,
	},
} );

export const setTicketEndDateInput = ( blockId, endDateInput ) => ( {
	type: types.SET_TICKET_END_DATE_INPUT,
	payload: {
		blockId,
		endDateInput,
	},
} );

export const setTicketEndDateMoment = ( blockId, endDateMoment ) => ( {
	type: types.SET_TICKET_END_DATE_MOMENT,
	payload: {
		blockId,
		endDateMoment,
	},
} );

export const setTicketStartTime = ( blockId, startTime ) => ( {
	type: types.SET_TICKET_START_TIME,
	payload: {
		blockId,
		startTime,
	},
} );

export const setTicketEndTime = ( blockId, endTime ) => ( {
	type: types.SET_TICKET_END_TIME,
	payload: {
		blockId,
		endTime,
	},
} );

export const setTicketStartTimeInput = ( blockId, startTimeInput ) => ( {
	type: types.SET_TICKET_START_TIME_INPUT,
	payload: {
		blockId,
		startTimeInput,
	},
} );

export const setTicketEndTimeInput = ( blockId, endTimeInput ) => ( {
	type: types.SET_TICKET_END_TIME_INPUT,
	payload: {
		blockId,
		endTimeInput,
	},
} );

export const setTicketCapacityType = ( blockId, capacityType ) => ( {
	type: types.SET_TICKET_CAPACITY_TYPE,
	payload: {
		blockId,
		capacityType,
	},
} );

export const setTicketCapacity = ( blockId, capacity ) => ( {
	type: types.SET_TICKET_CAPACITY,
	payload: {
		blockId,
		capacity,
	},
} );

//
// ─── TICKET TEMP DETAILS ACTIONS ────────────────────────────────────────────────
//

export const setTicketTempTitle = ( blockId, title ) => ( {
	type: types.SET_TICKET_TEMP_TITLE,
	payload: {
		blockId,
		title,
	},
} );

export const setTicketTempDescription = ( blockId, description ) => ( {
	type: types.SET_TICKET_TEMP_DESCRIPTION,
	payload: {
		blockId,
		description,
	},
} );

export const setTicketTempPrice = ( blockId, price ) => ( {
	type: types.SET_TICKET_TEMP_PRICE,
	payload: {
		blockId,
		price,
	},
} );

export const setTicketTempSku = ( blockId, sku ) => ( {
	type: types.SET_TICKET_TEMP_SKU,
	payload: {
		blockId,
		sku,
	},
} );

export const setTicketTempStartDate = ( blockId, startDate ) => ( {
	type: types.SET_TICKET_TEMP_START_DATE,
	payload: {
		blockId,
		startDate,
	},
} );

export const setTicketTempStartDateInput = ( blockId, startDateInput ) => ( {
	type: types.SET_TICKET_TEMP_START_DATE_INPUT,
	payload: {
		blockId,
		startDateInput,
	},
} );

export const setTicketTempStartDateMoment = ( blockId, startDateMoment ) => ( {
	type: types.SET_TICKET_TEMP_START_DATE_MOMENT,
	payload: {
		blockId,
		startDateMoment,
	},
} );

export const setTicketTempEndDate = ( blockId, endDate ) => ( {
	type: types.SET_TICKET_TEMP_END_DATE,
	payload: {
		blockId,
		endDate,
	},
} );

export const setTicketTempEndDateInput = ( blockId, endDateInput ) => ( {
	type: types.SET_TICKET_TEMP_END_DATE_INPUT,
	payload: {
		blockId,
		endDateInput,
	},
} );

export const setTicketTempEndDateMoment = ( blockId, endDateMoment ) => ( {
	type: types.SET_TICKET_TEMP_END_DATE_MOMENT,
	payload: {
		blockId,
		endDateMoment,
	},
} );

export const setTicketTempStartTime = ( blockId, startTime ) => ( {
	type: types.SET_TICKET_TEMP_START_TIME,
	payload: {
		blockId,
		startTime,
	},
} );

export const setTicketTempEndTime = ( blockId, endTime ) => ( {
	type: types.SET_TICKET_TEMP_END_TIME,
	payload: {
		blockId,
		endTime,
	},
} );

export const setTicketTempStartTimeInput = ( blockId, startTimeInput ) => ( {
	type: types.SET_TICKET_TEMP_START_TIME_INPUT,
	payload: {
		blockId,
		startTimeInput,
	},
} );

export const setTicketTempEndTimeInput = ( blockId, endTimeInput ) => ( {
	type: types.SET_TICKET_TEMP_END_TIME_INPUT,
	payload: {
		blockId,
		endTimeInput,
	},
} );

export const setTicketTempCapacityType = ( blockId, capacityType ) => ( {
	type: types.SET_TICKET_TEMP_CAPACITY_TYPE,
	payload: {
		blockId,
		capacityType,
	},
} );

export const setTicketTempCapacity = ( blockId, capacity ) => ( {
	type: types.SET_TICKET_TEMP_CAPACITY,
	payload: {
		blockId,
		capacity,
	},
} );

//
// ─── TICKET ACTIONS ─────────────────────────────────────────────────────────────
//

export const registerTicketBlock = ( blockId ) => ( {
	type: types.REGISTER_TICKET_BLOCK,
	payload: {
		blockId,
	},
} );

export const removeTicketBlock = ( blockId ) => ( {
	type: types.REMOVE_TICKET_BLOCK,
	payload: {
		blockId,
	},
} );

export const setTicketSold = ( blockId, sold ) => ( {
	type: types.SET_TICKET_SOLD,
	payload: {
		blockId,
		sold,
	},
} );

export const setTicketAvailable = ( blockId, available ) => ( {
	type: types.SET_TICKET_AVAILABLE,
	payload: {
		blockId,
		available,
	},
} );

export const setTicketId = ( blockId, ticketId ) => ( {
	type: types.SET_TICKET_ID,
	payload: {
		blockId,
		ticketId,
	},
} );

export const setTicketCurrencySymbol = ( blockId, currencySymbol ) => ( {
	type: types.SET_TICKET_CURRENCY_SYMBOL,
	payload: {
		blockId,
		currencySymbol,
	},
} );

export const setTicketCurrencyPosition = ( blockId, currencyPosition ) => ( {
	type: types.SET_TICKET_CURRENCY_POSITION,
	payload: {
		blockId,
		currencyPosition,
	},
} );

export const setTicketProvider = ( blockId, provider ) => ( {
	type: types.SET_TICKET_PROVIDER,
	payload: {
		blockId,
		provider,
	},
} );

export const setTicketIsLoading = ( blockId, isLoading ) => ( {
	type: types.SET_TICKET_IS_LOADING,
	payload: {
		blockId,
		isLoading,
	},
} );

export const setTicketHasBeenCreated = ( blockId, hasBeenCreated ) => ( {
	type: types.SET_TICKET_HAS_BEEN_CREATED,
	payload: {
		blockId,
		hasBeenCreated,
	},
} );

export const setTicketHasChanges = ( blockId, hasChanges ) => ( {
	type: types.SET_TICKET_HAS_CHANGES,
	payload: {
		blockId,
		hasChanges,
	},
} );

export const setTicketIsSelected = ( blockId, isSelected ) => ( {
	type: types.SET_TICKET_IS_SELECTED,
	payload: {
		blockId,
		isSelected,
	},
} );

//
// ─── TICKET SAGA ACTIONS ────────────────────────────────────────────────────────
//

export const setTicketDetails = ( blockId, details ) => ( {
	type: types.SET_TICKET_DETAILS,
	payload: {
		blockId,
		details,
	},
} );

export const setTicketTempDetails = ( blockId, tempDetails ) => ( {
	type: types.SET_TICKET_TEMP_DETAILS,
	payload: {
		blockId,
		tempDetails,
	},
} );

export const handleTicketStartDate = ( blockId, date, dayPickerInput ) => ( {
	type: types.HANDLE_TICKET_START_DATE,
	payload: {
		blockId,
		date,
		dayPickerInput,
	},
} );

export const handleTicketEndDate = ( blockId, date, dayPickerInput ) => ( {
	type: types.HANDLE_TICKET_END_DATE,
	payload: {
		blockId,
		date,
		dayPickerInput,
	},
} );

export const handleTicketStartTime = ( blockId, seconds ) => ( {
	type: types.HANDLE_TICKET_START_TIME,
	payload: {
		blockId,
		seconds,
	},
} );

export const handleTicketEndTime = ( blockId, seconds ) => ( {
	type: types.HANDLE_TICKET_END_TIME,
	payload: {
		blockId,
		seconds,
	},
} );


export const fetchTicket = ( blockId, ticketId ) => ( {
	type: types.FETCH_TICKET,
	payload: {
		blockId,
		ticketId,
	},
} );

export const createNewTicket = ( blockId ) => ( {
	type: types.CREATE_NEW_TICKET,
	payload: {
		blockId,
	},
} );

export const updateTicket = ( blockId ) => ( {
	type: types.UPDATE_TICKET,
	payload: {
		blockId,
	},
} );

export const deleteTicket = ( blockId ) => ( {
	type: types.DELETE_TICKET,
	payload: {
		blockId,
	}
} );

export const setTicketInitialState = ( props ) => ( {
	type: types.SET_TICKET_INITIAL_STATE,
	payload: props,
} );
