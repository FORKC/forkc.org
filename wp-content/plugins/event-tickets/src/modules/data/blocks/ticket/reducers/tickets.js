/**
 * External dependencies
 */
import { combineReducers } from 'redux';
import omit from 'lodash/omit';

/**
 * Internal dependencies
 */
import * as types from '@moderntribe/tickets/data/blocks/ticket/types';
import ticket from './tickets/ticket';

export const byId = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.SET_TICKET_TITLE:
		case types.SET_TICKET_DESCRIPTION:
		case types.SET_TICKET_PRICE:
		case types.SET_TICKET_SKU:
		case types.SET_TICKET_START_DATE:
		case types.SET_TICKET_START_DATE_INPUT:
		case types.SET_TICKET_START_DATE_MOMENT:
		case types.SET_TICKET_END_DATE:
		case types.SET_TICKET_END_DATE_INPUT:
		case types.SET_TICKET_END_DATE_MOMENT:
		case types.SET_TICKET_START_TIME:
		case types.SET_TICKET_END_TIME:
		case types.SET_TICKET_START_TIME_INPUT:
		case types.SET_TICKET_END_TIME_INPUT:
		case types.SET_TICKET_CAPACITY_TYPE:
		case types.SET_TICKET_CAPACITY:
		case types.SET_TICKET_TEMP_TITLE:
		case types.SET_TICKET_TEMP_DESCRIPTION:
		case types.SET_TICKET_TEMP_PRICE:
		case types.SET_TICKET_TEMP_SKU:
		case types.SET_TICKET_TEMP_START_DATE:
		case types.SET_TICKET_TEMP_START_DATE_INPUT:
		case types.SET_TICKET_TEMP_START_DATE_MOMENT:
		case types.SET_TICKET_TEMP_END_DATE:
		case types.SET_TICKET_TEMP_END_DATE_INPUT:
		case types.SET_TICKET_TEMP_END_DATE_MOMENT:
		case types.SET_TICKET_TEMP_START_TIME:
		case types.SET_TICKET_TEMP_END_TIME:
		case types.SET_TICKET_TEMP_START_TIME_INPUT:
		case types.SET_TICKET_TEMP_END_TIME_INPUT:
		case types.SET_TICKET_TEMP_CAPACITY_TYPE:
		case types.SET_TICKET_TEMP_CAPACITY:
		case types.SET_TICKET_SOLD:
		case types.SET_TICKET_AVAILABLE:
		case types.SET_TICKET_ID:
		case types.SET_TICKET_CURRENCY_SYMBOL:
		case types.SET_TICKET_CURRENCY_POSITION:
		case types.SET_TICKET_PROVIDER:
		case types.SET_TICKET_IS_LOADING:
		case types.SET_TICKET_HAS_BEEN_CREATED:
		case types.SET_TICKET_HAS_CHANGES:
		case types.SET_TICKET_IS_SELECTED:
		case types.REGISTER_TICKET_BLOCK:
			return {
				...state,
				[ action.payload.blockId ]: ticket( state[ action.payload.blockId ], action ),
			};
		case types.REMOVE_TICKET_BLOCK:
			return omit( state, [ action.payload.blockId ] );
		default:
			return state;
	}
};

export const allIds = ( state = [], action ) => {
	switch ( action.type ) {
		case types.REGISTER_TICKET_BLOCK:
			return [ ...state, action.payload.blockId ];
		case types.REMOVE_TICKET_BLOCK:
			return state.filter( ( id ) => action.payload.blockId !== id );
		default:
			return state;
	}
};

export default combineReducers( {
	byId,
	allIds,
} );
