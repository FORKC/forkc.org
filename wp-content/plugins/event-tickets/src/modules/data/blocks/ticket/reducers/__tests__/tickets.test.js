/**
 * Internal dependencies
 */
import tickets, { byId, allIds } from '../tickets';
import { actions } from '@moderntribe/tickets/data/blocks/ticket';

jest.mock( 'moment', () => () => {
	const moment = require.requireActual( 'moment' );
	return moment( 'January 10, 2018 5:30 pm', 'MMMM D, Y h:mm a' );
} );

describe( 'Tickets reducer', () => {
	describe( 'byId', () => {
		it( 'should set the default state', () => {
			expect( byId( undefined, {} ) ).toEqual( {} );
		} );

		it( 'should register ticket block', () => {
			expect( byId(
				{},
				actions.registerTicketBlock( 'tribe' ),
			) ).toMatchSnapshot();
		} );

		it( 'should remove ticket block', () => {
			expect( byId(
				{ one: {}, two: {} },
				actions.removeTicketBlock( 'one' ),
			) ).toMatchSnapshot();
		} );
	} );

	describe( 'allIds', () => {
		it( 'should set the default state', () => {
			expect( allIds( undefined, {} ) ).toMatchSnapshot();
		} );

		it( 'should register ticket block', () => {
			expect( allIds(
				[],
				actions.registerTicketBlock( 'tribe' ),
			) ).toMatchSnapshot();
		} );

		it( 'should remove ticket block', () => {
			expect( allIds(
				[ 'one', 'two' ],
				actions.removeTicketBlock( 'one' ),
			) ).toMatchSnapshot();
		} );
	} );

	describe( 'tickets', () => {
		it( 'should set the default state', () => {
			expect( tickets( undefined, {} ) ).toMatchSnapshot();
		} );
	} );
} );
