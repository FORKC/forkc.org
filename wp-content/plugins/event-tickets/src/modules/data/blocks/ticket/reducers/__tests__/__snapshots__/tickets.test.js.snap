// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tickets reducer allIds should register ticket block 1`] = `
Array [
  "tribe",
]
`;

exports[`Tickets reducer allIds should remove ticket block 1`] = `
Array [
  "two",
]
`;

exports[`Tickets reducer allIds should set the default state 1`] = `Array []`;

exports[`Tickets reducer byId should register ticket block 1`] = `
Object {
  "tribe": Object {
    "available": 0,
    "currencyPosition": "prefix",
    "currencySymbol": undefined,
    "details": Object {
      "capacity": "",
      "capacityType": "capped",
      "description": "",
      "endDate": "2118-01-10",
      "endDateInput": "January 10, 2118",
      "endDateMoment": "2118-01-10T17:30:00.000Z",
      "endTime": "17:30:00",
      "endTimeInput": "5:30 pm",
      "price": "",
      "sku": "",
      "startDate": "2018-01-10",
      "startDateInput": "January 10, 2018",
      "startDateMoment": "2018-01-10T17:30:00.000Z",
      "startTime": "17:30:00",
      "startTimeInput": "5:30 pm",
      "title": "",
    },
    "hasBeenCreated": false,
    "hasChanges": false,
    "isLoading": false,
    "isSelected": false,
    "provider": "",
    "sold": 0,
    "tempDetails": Object {
      "capacity": "",
      "capacityType": "capped",
      "description": "",
      "endDate": "2118-01-10",
      "endDateInput": "January 10, 2118",
      "endDateMoment": "2118-01-10T17:30:00.000Z",
      "endTime": "17:30:00",
      "endTimeInput": "5:30 pm",
      "price": "",
      "sku": "",
      "startDate": "2018-01-10",
      "startDateInput": "January 10, 2018",
      "startDateMoment": "2018-01-10T17:30:00.000Z",
      "startTime": "17:30:00",
      "startTimeInput": "5:30 pm",
      "title": "",
    },
    "ticketId": 0,
  },
}
`;

exports[`Tickets reducer byId should remove ticket block 1`] = `
Object {
  "two": Object {},
}
`;

exports[`Tickets reducer tickets should set the default state 1`] = `
Object {
  "allIds": Array [],
  "byId": Object {},
}
`;
